tests/testthat/test-formatters.R

```diff
@@ -551,3 +551,160 @@ mpf <- MatrixPrintForm(strings = strs, spans = spans, aligns = aligns,
 
 expect_equal(length(grep("spn_val", toString(mpf))),
              1L)
+
+# testing if error is returned when provided widths argument
+# is narrower than decimal aligned values
+
+df_error <- dfmf
+df_error$aligns[, -c(1)] <- "dec_left"
+
+expect_error(
+  toString(df_error, widths = c(25, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4)),
+  "Inserted width\\(s\\) for column\\(s\\) disp, wt, qsec is\\(are\\) not wide enough for the desired alignment"
+)
+
+
+# test if padstr works with "dec_left", "dec_right" and "decimal"
+
+test_that("padstr works with dec_left", {
+  bmf <- basic_matrix_form(mtcars[1:15, c(1, 6)])
+  bmf$aligns[, -c(1)] <- "dec_left"
+  result <- strsplit(toString(bmf), "\\n")[[1]]
+  expected <- c(
+    "                     mpg    wt   ",
+    "—————————————————————————————————",
+    "Mazda RX4            21     2.62 ",
+    "Mazda RX4 Wag        21     2.875",
+    "Datsun 710           22.8   2.32 ",
+    "Hornet 4 Drive       21.4   3.215",
+    "Hornet Sportabout    18.7   3.44 ",
+    "Valiant              18.1   3.46 ",
+    "Duster 360           14.3   3.57 ",
+    "Merc 240D            24.4   3.19 ",
+    "Merc 230             22.8   3.15 ",
+    "Merc 280             19.2   3.44 ",
+    "Merc 280C            17.8   3.44 ",
+    "Merc 450SE           16.4   4.07 ",
+    "Merc 450SL           17.3   3.73 ",
+    "Merc 450SLC          15.2   3.78 ",
+    "Cadillac Fleetwood   10.4   5.25 "
+  )
+  expect_identical(result, expected)
+})
+
+test_that("padstr works with dec_right", {
+  bmf <- basic_matrix_form(mtcars[1:15, c(1, 6)])
+  bmf$aligns[, -c(1)] <- "dec_right"
+  result <- strsplit(toString(bmf), "\\n")[[1]]
+  expected <- c(
+      "                      mpg      wt",
+      "—————————————————————————————————",
+      "Mazda RX4              21   2.62 ",
+      "Mazda RX4 Wag          21   2.875",
+      "Datsun 710           22.8   2.32 ",
+      "Hornet 4 Drive       21.4   3.215",
+      "Hornet Sportabout    18.7   3.44 ",
+      "Valiant              18.1   3.46 ",
+      "Duster 360           14.3   3.57 ",
+      "Merc 240D            24.4   3.19 ",
+      "Merc 230             22.8   3.15 ",
+      "Merc 280             19.2   3.44 ",
+      "Merc 280C            17.8   3.44 ",
+      "Merc 450SE           16.4   4.07 ",
+      "Merc 450SL           17.3   3.73 ",
+      "Merc 450SLC          15.2   3.78 ",
+      "Cadillac Fleetwood   10.4   5.25 "
+    )
+    expect_identical(result, expected)
+})
+
+test_that("padstr works with dec_right", {
+  bmf <- basic_matrix_form(mtcars[1:15, c(1, 6)])
+  bmf$aligns[, -c(1)] <- "decimal"
+  result <- strsplit(toString(bmf), "\\n")[[1]]
+  expected <- c(
+    "                     mpg     wt  ",
+    "—————————————————————————————————",
+    "Mazda RX4             21    2.62 ",
+    "Mazda RX4 Wag         21    2.875",
+    "Datsun 710           22.8   2.32 ",
```

Comment:
I think if you see [here](https://user-images.githubusercontent.com/3692541/167969307-3347566a-4e92-41a3-a7a9-26cd279ba93e.png) the `.` (dot) should stay on the right of non-dot values e.g. here it should be:

```
  21
  21
  22.8
234.9
```
which would be different from the `dec_left`:
```
21
21
  22.8
234.9
```